ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ActiveContractsServiceIT.scala

```diff
@@ -394,6 +400,340 @@ class ActiveContractsServiceIT extends LedgerTestSuite {
     )
   })
 
+  test(
+    "ACSFilterCombinations",
+    "Testing a few combinations of ACSs",
+    allocate(Parties(3)),
+  )(implicit ec => { case Participants(Participant(ledger, p1, p2, p3)) =>
+    // Let we have 3 templates
+    val templateIds: Vector[Identifier] =
+      Vector(TriAgreement.id.unwrap, TriProposal.id.unwrap, WithObservers.id.unwrap)
+    // Let we have 3 parties
+    val parties: Vector[Party] = Vector(p1, p2, p3)
+    // Let we have all combinations for the 3 parties
+    val partyCombinations =
+      Vector(Set(0), Set(1), Set(2), Set(0, 1), Set(1, 2), Set(0, 2), Set(0, 1, 2))
+    case class FilterCoord(templateId: Int, stakeholders: Set[Int])
+    // Let we populate 3 contracts for each template/partyCombination pair (see createContracts below)
+    // Then we require the following Filter - Expectations to be upheld
+    val * = Set.empty[Int]
+    val allFilterCoords = Set(
+      FilterCoord(0, Set(0)),
+      FilterCoord(0, Set(1)),
+      FilterCoord(0, Set(2)),
+      FilterCoord(0, Set(0, 1)),
+      FilterCoord(0, Set(0, 2)),
+      FilterCoord(0, Set(1, 2)),
+      FilterCoord(0, Set(0, 1, 2)),
+      FilterCoord(1, Set(0)),
+      FilterCoord(1, Set(1)),
+      FilterCoord(1, Set(2)),
+      FilterCoord(1, Set(0, 1)),
+      FilterCoord(1, Set(0, 2)),
+      FilterCoord(1, Set(1, 2)),
+      FilterCoord(1, Set(0, 1, 2)),
+      FilterCoord(2, Set(0)),
+      FilterCoord(2, Set(1)),
+      FilterCoord(2, Set(2)),
+      FilterCoord(2, Set(0, 1)),
+      FilterCoord(2, Set(0, 2)),
+      FilterCoord(2, Set(1, 2)),
+      FilterCoord(2, Set(0, 1, 2)),
+    )
+    val fixtures: Vector[(Map[Int, Set[Int]], Set[FilterCoord])] = Vector(
+      // single filter
+      Map(0 -> *) -> Set(
```

Comment:
I am a bit reluctant adding too much logic to fixture generation, but since all of you agree, I am happy to do so.
@kamil-da thanks for the approach! FYI that is actually changing the wildcard template filters back to concrete filters, which is not what we want to test there. I adapted your code :+1:

Reply:
> Moreover, if you would like to make assertions about the results of this test-implemention of filter execution, then this could be done as a normal assertion.

@meiersi-da can you please elaborate?